src/AWS.Distro.OpenTelemetry.AutoInstrumentation/LambdaWrapper.cs

```diff
@@ -0,0 +1,188 @@
+    private async Task<object?> FunctionHandler(JObject input, ILambdaContext context)
+    {
+        if (input == null)
+        {
+            throw new Exception($"Input cannot be null.");
+        }
+
+        (MethodInfo handlerMethod, object handlerInstance) = this.ExtractOriginalHandler();
+
+        object? originalHandlerResult;
+        ParameterInfo[] parameters = handlerMethod.GetParameters();
+
+        // A .NET Lambda handler function can have zero, one, or two parameters, depending on the customer's needs:
+        //      * Zero Parameters:  When no input data or context is needed.
+        //      * One Parameter:    When only the event data is needed.
+        //      * Two Parameters:   When both the event data and execution context are needed.
+        if (parameters.Length == 2)
+        {
+            Type inputParameterType = parameters[0].ParameterType;
+            object? inputObject = input.ToObject(inputParameterType);
+
+            if (inputObject == null)
+            {
+                throw new Exception($"Wrapper wasn't able to convert the input object to type: {inputParameterType}!");
+            }
+
+            originalHandlerResult = handlerMethod.Invoke(handlerInstance, new object[] { inputObject, context });
+        }
+        else if (parameters.Length == 1)
+        {
+            Type inputParameterType = parameters[0].ParameterType;
+            object? inputObject = input.ToObject(inputParameterType);
+
+            if (inputObject == null)
+            {
+                throw new Exception($"Wrapper wasn't able to convert the input object to type: {inputParameterType}!");
+            }
+
+            originalHandlerResult = handlerMethod.Invoke(handlerInstance, new object[] { inputObject });
+        }
+        else if (parameters.Length == 0)
+        {
+            originalHandlerResult = handlerMethod.Invoke(handlerInstance, new object[] { });
+        }
+        else
+        {
+            throw new Exception($"Wrapper handler doesn't support more than 2 input paramaters");
+        }
+
+        Type returnType = handlerMethod.ReturnType;
+        if (originalHandlerResult == null && returnType.ToString() != typeof(void).ToString())
+        {
+            throw new Exception($"originalHandlerResult of type: {returnType} returned from the original handler is null!");
+        }
+
+        // AsyncStateMachineAttribute can be used to determine whether the original handler method is
+        // asynchronous vs synchronous
+        Type asyncAttribType = typeof(AsyncStateMachineAttribute);
+        var asyncAttrib = (AsyncStateMachineAttribute?)handlerMethod.GetCustomAttribute(asyncAttribType);
+
+        // The return type of the original lambda function is Task<T> or Task so we await for it
+        if (asyncAttrib != null && originalHandlerResult != null)
+        {
+            await (Task)originalHandlerResult;
```

Comment:
It doesn't timeout, it will basically wait indefinitely until it returns. It's hard to choose a timeout as this depends on the customer function and how big it is.